akka-projection-grpc/src/main/scala/akka/projection/grpc/producer/scaladsl/EventProducer.scala

```diff
@@ -152,6 +152,10 @@ object EventProducer {
       registerAsyncMapper[A, B](event => Future.successful(f(event)))
     }
 
+    def registerEnvelopeMapper[A: ClassTag, B](f: EventEnvelope[A] => Option[B]): Transformation = {
```

Comment:
It's a bit of a challenge to use the same naming between the different transformers, where it makes sense. We also have consumer transformation with `registerPayloadMapper`.

We have registerEnvelopeMapper in the original (released) producer transformation, so we should probably stick to that naming. Meaning that we should rename `registerPayloadMapper`?

Reply:
Aligned names and added some more to make the two transformer APIs looks mostly the same. Added API docs as well.